client/python/gradio_client/client.py

```diff
@@ -603,6 +605,135 @@ def _get_config(self) -> dict:
                 )
             return config
 
+    def deploy_discord(
+        self,
+        discord_bot_token: str | None = None,
+        api_names: list[str] | None = None,
+        command_names: list[str] | None = None,
+        to_id: str | None = None,
+        hf_token: str | None = None,
+        private: bool = True,
+        persist_state: bool = False,
+    ):
+        """
+        Deploy the upstream app as a discord bot. Currently only supports gr.ChatInterface.
+        Parameters:
+            discord_bot_token: This is the "password" needed to be able to launch the bot. Users can get a token by creating a bot app on the discord website. If run the method without specifying a token, the space will explain how to get one. See here: https://huggingface.co/spaces/freddyaboulton/test-discord-bot-v1.
+            api_names: The api_names of the app to turn into bot commands. This parameter currently has no effect as ChatInterface only has one api_name ('/chat').
+            command_names: The names of the command the bot will have. Must be the same length as api_names. For gr.ChatInterface, this list must contain one element. For example, if the value is ['bot'], the bot command can be triggered via !bot.
+            to_id: The name of the space hosting the discord bot. If None, the name will be gradio-discord-bot-{random-substring}
```

Comment:
Nit: maybe it should reflect the name of the original Space as well? `gradio-discord-bot-{space-id}-{random-substring}`

Reply:
Fixed this - the name of the bot space will use the upstream space name and add `gradio-discord-bot` if `to_id` is not passed in.